### PR TITLE
Initial v2 - much simpler

### PR DESCRIPTION
**Pull request by joshtwist**
https://www.loom.com/share/6f68a06ec380494f84b6625dfd65df82

### CODE REVIEW

File: examples/nextjs/.env.local [hunk 1]
```diff
@@ -1,4 +1,4 @@
-NEXT_PUBLIC_API_URL=
+NEXT_PUBLIC_API_URL=https://sample-auth-translation-api-main-d1b33d3.d2.zuplo.dev
```
Comment (AdrianMachado) [inline examples/nextjs/.env.local:1]: Don't commit this

File: packages/react/src/components/ApiKeyManager.tsx [hunk 2]
```diff
@@ -24,30 +25,66 @@ const getSystemDefaultThemePreference = (): "dark" | "light" => {
...
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
+      setError(err.message);
+      console.error(err);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    loadData(provider);
```
Comment (AdrianMachado) [inline packages/react/src/components/ApiKeyManager.tsx:60]: If you're just trying to make an async call from this you can do
```suggestion
    void loadData(provider);
```

File: packages/react/src/components/ConsumerControl.tsx [hunk 3]
```diff
@@ -12,73 +11,93 @@ import {
...
+      await provider.updateConsumerDescription(consumer.name, description);
+      const result = await provider.getConsumers();
+      setDataModel({
+        isFetching: dataModel?.isFetching,
+        consumers: result.data,
+      });
+
+      setEdit(false);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
```
Comment (AdrianMachado) [inline packages/react/src/components/ConsumerControl.tsx:54]: You don't need to actually add the type here
```suggestion
    } catch (err) {
```

File: packages/react/src/components/ConsumerControl.tsx [hunk 4]
```diff
@@ -12,73 +11,93 @@ import {
...
+      setDescriptionUpdating(true);
+      await provider.updateConsumerDescription(consumer.name, description);
+      const result = await provider.getConsumers();
+      setDataModel({
+        isFetching: dataModel?.isFetching,
+        consumers: result.data,
+      });
+
+      setEdit(false);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (AdrianMachado) [inline packages/react/src/components/ConsumerControl.tsx:53]: Should we clear the error after a successful request?
```suggestion
      setError(undefined);
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

File: packages/react/src/components/CreateConsumer.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        onClick={() => setEditMode(true)}
+        className="text-white rounded bg-pink-500 hover:bg-pink-700 p-2 px-4"
+      >
+        Create new API Key
+      </button>
+    );
+  }
+
+  return (
+    <div className="flex flex-row items-center gap-x-2 w-full">
```
Comment (AdrianMachado) [inline packages/react/src/components/CreateConsumer.tsx:61]: No inline tailwind anymore - you have to do it through a css file like the other files do. Its a bit tedious to do so I usually ask chatGPT to extract them for me :)

**Review by AdrianMachado — CHANGES_REQUESTED** [review]
Good cleanup. Left some questions for you. CSS needs to be cleaned up into split file

File: packages/react/src/components/ConsumerControl.tsx (hunk 3 above)
Comment (joshtwist) [reply packages/react/src/components/ConsumerControl.tsx:57]: I can't use the .message property if I don't specify the type :(

**Review by AdrianMachado — APPROVED** [review]

**Pull request merged**